pkg/roachpb/api.proto

```diff
@@ -1762,6 +1762,12 @@ message AddSSTableRequest {
   // the usual write pipeline (on-disk raft log, WAL, etc).
   // TODO(dt): https://github.com/cockroachdb/cockroach/issues/34579#issuecomment-544627193
   bool ingest_as_writes = 5;
+
+  // CheckForDataAbove, if set, instructs the request to additionally inspect 
+  // the span between the end of the added file and the end of the range to 
+  // which it was added and return the lowest key, if any, at which that span 
+  // is non-empty in the response field NonEmptyAbove.
+  bool check_for_data_above = 9;
```

Comment:
nit: to me, above sounds like across the span at higher timestamps. After? Following? RHS?

Reply:
ReturnFollowingNonEmptySpanStart ?

Reply:
or maybe `ReturnFollowingLikelyNonEmptySpanStart` to emphasize the reduced promise we're making w.r.t. consistency and isolation